src/model.cpp

```diff
@@ -240,6 +243,41 @@ void Model::initializeStates(AmiVector &x) {
     }
 }
 
+void Model::initializeSplines() {
+    splines_ = fspline_constructors(state_.unscaledParameters.data(),
+                                    state_.fixedParameters.data());
+    state_.spl_.resize(splines_.size(), 0.0);
+    for(auto& spline: splines_) {
+        spline.compute_coefficients();
+    }
+}
+
+void Model::initializeSplineSensitivities() {
+    int nspl = splines_.size();
+    derived_state_.sspl_ = SUNMatrixWrapper(nspl, nplist());
+    int allnodes = 0;
+    for(auto const& spline: splines_) {
+        allnodes += spline.n_nodes();
+    }
+
+    std::vector<realtype> dspline_valuesdp (allnodes * nplist(), 0.0);
+    std::vector<realtype> dspline_slopesdp (allnodes * nplist(), 0.0);
+    fdspline_valuesdp(dspline_valuesdp.data(), state_.unscaledParameters.data(),
+                      state_.fixedParameters.data());
+    fdspline_slopesdp(dspline_slopesdp.data(), state_.unscaledParameters.data(),
+                      state_.fixedParameters.data());
+
+    // QUESTION is it possible for plist != range(nplist) ?
+    //          would that break computeCoefficientsSensi or not?
```

Comment:
Yes, `plist` will often not be equal to `range(nplist)`. Seems like a good test case...